doc/docusaurus.config.js

```diff
@@ -83,7 +83,7 @@ const config = {
       /** @type {import('@docusaurus/preset-classic').Options} */
       ({
         docs: {
-          remarkPlugins: [require('mdx-mermaid')],
+          remarkPlugins: [import('mdx-mermaid')],
```

Comment:
Docusaurus does not support ESM yet https://github.com/facebook/docusaurus/issues/5379
I am happy to merge this change in (and make the appropriate changes to document this) if there are people wanting to use this library outside of Docusaurus.
I don't see a reason to merge this change until Docusaurus is up upgraded to support ESM if there are no use cases outside of it. The main reason for this is that it could cause confusion in the documentation and will add more support overhead

Reply:
Are you sure that doesn't work?
The GitHub Action at https://github.com/candrews/mdx-mermaid/blob/135b622f14972e347b1e6b84f9303d3ea23cee43/.github/workflows/build.yml#L28 passes, meaning that the "doc" project builds (which includes the `require` -> `import` change).

Reply:
The code compiles and the React component works but the mdx parser in Docusuarus doesn't process the markdown files.
Running up the docs locally with `yarn start`  at the bottom of intro.md there should be a diagram http://localhost:3000/mdx-mermaid/docs/intro/

Actual:
![image](https://user-images.githubusercontent.com/16527946/146650996-c4820428-db57-4726-b791-3dc41ee36d4a.png)

Expected:
![image](https://user-images.githubusercontent.com/16527946/146651008-d7dcbb5b-a5e9-49c4-be5d-5198cb269a6d.png)

The diagram on the index works because it is using the React component directly and not the plugin https://github.com/sjwall/mdx-mermaid/blob/3d6784ca2d18ab59a302d46d3ed3c69fdeeee5ff/doc/src/components/HomepageFeatures.js#L10-L19

The import statement is returning a promise which I guess that the parser is ignoring as it doesn't know how to handle it

Reply:
A dynamic import is always a promise this is why it needs to be awaited. Docusaurus itself is also using `.mjs` inside their [own config](https://github.com/facebook/docusaurus/blob/main/website/docusaurus.config.js#L556-L559).

Which means the docusaurus config needs to be following: 

```js
const createConfig = async () => ({
  ...
  docs: {
    remarkPlugins: [
      (await import('mdx-mermaid')).default,
    ],
  },
  ...
});

module.exports = createConfig;
```

With that config it should actually work.